routes/campgrounds.js

```diff
@@ -27,35 +27,55 @@ cloudinary.config({
 
 // INDEX - show all campgrounds in the web
 router.get('/', (req, res) => {
+  const perPage = 8;
+  const pageQuery = parseInt(req.query.page);
+  const pageNumber = pageQuery ? pageQuery : 1;
   let noMatch = null;
   if (req.query.search) {
     const regex = new RegExp(escapeRegex(req.query.search), 'gi');
     // Get all campgrounds from DB
-    Campground.find({ name: regex }, (err, allCampgrounds) => {
-      if (err) {
-        console.log(err);
-      } else {
-        if (!allCampgrounds.length) {
-          noMatch = `No campgrounds match "${req.query.search}" query, please try again.`;
-        }
-        res.render('campgrounds/index', {
-          campgrounds: allCampgrounds,
-          noMatch: noMatch,
+    Campground.find({ name: regex })
+      .skip(perPage * pageNumber - perPage)
+      .limit(perPage)
+      .exec((err, allCampgrounds) => {
+        Campground.count({ name: regex }).exec((err, count) => {
+          if (err) {
+            console.log(err);
+            res.redirect('back');
+          } else {
+            if (!allCampgrounds.length) {
+              noMatch = `No campgrounds match "${req.query.search}" query, please try again.`;
+            }
+            res.render('campgrounds/index', {
+              campgrounds: allCampgrounds,
+              current: pageNumber,
+              pages: Math.ceil(count / perPage),
+              noMatch: noMatch,
+              search: req.query.search,
+            });
+          }
         });
-      }
-    });
+      });
   } else {
     // Get all campgrounds from DB
-    Campground.find({}, function (err, allCampgrounds) {
-      if (err) {
-        console.log(err);
-      } else {
-        res.render('campgrounds/index', {
-          campgrounds: allCampgrounds,
-          noMatch: noMatch,
+    Campground.find({})
+      .skip(perPage * pageNumber - perPage)
```

Comment:
You could set `perPage * pageNumber - perPage` to some variable and reuse it in both places you have this expression.